tests/functional/spec/numbers.spec.js

```diff
@@ -1,62 +1,84 @@
-    expect($(TestMinMax.errorNumber(4)).getText()).to.contain("Enter an answer less than or equal to 1,234");
-    expect($(TestMinMax.errorNumber(5)).getText()).to.contain("Enter an answer more than 123");
-    expect($(TestMinMax.errorNumber(6)).getText()).to.contain("Enter an answer less than 1,234");
-    expect($(TestMinMax.errorNumber(7)).getText()).to.contain("Enter an answer less than or equal to 100");
-    expect($(TestMinMax.errorNumber(8)).getText()).to.contain("Enter an answer more than or equal to £10.00");
-  });
+      expect($(TestMinMax.errorNumber(1)).getText()).to.contain("Enter an answer more than or equal to 10");
+      expect($(TestMinMax.errorNumber(2)).getText()).to.contain("Enter an answer more than 10");
+      expect($(TestMinMax.errorNumber(3)).getText()).to.contain("Enter an answer more than or equal to 123");
+      expect($(TestMinMax.errorNumber(4)).getText()).to.contain("Enter an answer less than or equal to 1,234");
+      expect($(TestMinMax.errorNumber(5)).getText()).to.contain("Enter an answer more than 123");
+      expect($(TestMinMax.errorNumber(6)).getText()).to.contain("Enter an answer less than 1,234");
+      expect($(TestMinMax.errorNumber(7)).getText()).to.contain("Enter an answer less than or equal to 100");
+      expect($(TestMinMax.errorNumber(8)).getText()).to.contain("Enter an answer more than or equal to £10.00");
+    });
+
+    it("When I enter values inside the set range but provide too many decimal places, Then the correct error messages are displayed", () => {
+      $(TestMinMax.testRange()).setValue("1020");
+      $(TestMinMax.testRangeExclusive()).setValue("11");
+      $(TestMinMax.testMin()).setValue("123");
+      $(TestMinMax.testMax()).setValue("1019");
+      $(TestMinMax.testMinExclusive()).setValue("124");
+      $(TestMinMax.testMaxExclusive()).setValue("1233");
+      $(TestMinMax.testPercent()).setValue("100");
+      $(TestMinMax.testRange()).setValue("12.344");
+      $(TestMinMax.testDecimal()).setValue("11.234");
+      $(TestMinMax.submit()).click();
+
+      expect($(TestMinMax.errorNumber(1)).getText()).to.contain("Enter a number rounded to 2 decimal places");
+      expect($(TestMinMax.errorNumber(2)).getText()).to.contain("Enter a number rounded to 2 decimal places");
+    });
+
+    it("When I enter values inside the set range, Then I should be able to submit the survey", () => {
+      $(TestMinMax.testRange()).setValue("12");
+      $(TestMinMax.testDecimal()).setValue("11.23");
+      $(TestMinMax.testPercent()).setValue("99");
+      $(TestMinMax.submit()).click();
+      $(DetailAnswer.other()).click();
+      $(DetailAnswer.otherDetail()).setValue("1020");
+      $(DetailAnswer.submit()).click();
+
+      expect(browser.getUrl()).to.contain(SubmitPage.pageName);
+    });
+
+    it("When I edit and change a question which has dependent answer, Then I must re-answer if required and re-submit before I can return to the summary", () => {
```

Comment:
We should test both min and max, they are two separate constructs, so to verify both works, we should test both.